xcodeproj/internal/pbxproj_partials.bzl

```diff
@@ -0,0 +1,78 @@
+"""Actions for creating `PBXProj` partials."""
+
+def _write_pbxproject_prefix(
+        *,
+        actions,
+        colorize,
+        execution_root_file,
+        generator_name,
+        minimum_xcode_version,
+        project_options,
+        tool,
+        workspace_directory):
+    """
+    Creates a `File` containing a `PBXProject` prefix `PBXProj` partial.
+
+    Args:
+        actions: `ctx.actions`.
+        colorize: A `bool` indicating whether to colorize the output.
+        execution_root_file: A `File` containing the absolute path to the Bazel
+            execution root.
+        generator_name: The name of the `xcodeproj` generator target.
+        minimum_xcode_version: The minimum Xcode version that the generated
+            project supports, as a `string`.
+        project_options: A `dict` as returned by `project_options`.
+        tool: The executable that will generate the pbxproj partial.
+        workspace_directory: The absolute path to the Bazel workspace
+            directory.
+
+    Returns:
+        The `File` for the `PBXProject` prefix `PBXProj` partial.
+    """
+    output = actions.declare_file(
+        "{}_pbxproj_partials/project_prefix".format(
+            generator_name,
+        ),
+    )
+
+    args = actions.args()
+
+    # outputPath
+    args.add(output)
+
+    # workspace
+    args.add(workspace_directory)
+
+    # executionRootFile
+    args.add(execution_root_file)
+
+    # minimumXcodeVersion
+    args.add(minimum_xcode_version)
+
+    # developmentRegion
+    development_region = project_options.get("development_region")
+    if development_region and development_region != "en":
+        args.add("--development-region", development_region)
+
+    # organizationName
+    organization_name = project_options.get("organization_name")
+    if organization_name:
+        args.add("--organization-name", organization_name)
+
+    # colorize
+    if colorize:
+        args.add("--colorize")
+
+    actions.run(
+        arguments = [args],
+        executable = tool,
+        inputs = [execution_root_file],
+        outputs = [output],
+        mnemonic = "WriteXcodeProjPBXProjectPrefixPBXProjPartial",
```

Comment:
Nit: This mnemonic seems to hold a lot of info? Do you think it's important to add all these? What about `WritePBXProjPrefix` since `PBXProj` already implies everything written in [tools/generators/README.md](https://github.com/MobileNativeFoundation/rules_xcodeproj/blob/main/tools/generators/README.md)?

Reply:
Yeah, I can shorten it.

Reply:
Went with `WritePBXProjPBXProjectPrefix`. All of the partials will use the `WritePBXProj` prefix, so they can get modified with `--modify_execution_info` easily. Then they will have a suffix that matches their generator name.